packages/espressocash_app/lib/features/stellar/service/recovery_service.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:decimal/decimal.dart';
+import 'package:espressocash_api/espressocash_api.dart';
+import 'package:flutter/foundation.dart';
+import 'package:injectable/injectable.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:stellar_flutter_sdk/stellar_flutter_sdk.dart' hide Currency;
+
+import '../../accounts/auth_scope.dart';
+import '../../accounts/models/account.dart';
+import '../../accounts/models/ec_wallet.dart';
+import '../../currency/models/amount.dart';
+import '../../currency/models/currency.dart';
+import '../models/stellar_wallet.dart';
+import 'stellar_client.dart';
+
+@Singleton(scope: authScope)
+class StellarRecoveryService extends ChangeNotifier {
+  StellarRecoveryService(
+    this._ecWallet,
+    this._stellarWallet,
+    this._stellarClient,
+    this._ecClient,
+    this._account,
+    this._storage,
+  );
+
+  final MyAccount _account;
+  final ECWallet _ecWallet;
+  final StellarWallet _stellarWallet;
+
+  final StellarClient _stellarClient;
+  final EspressoCashClient _ecClient;
+
+  final SharedPreferences _storage;
+
+  bool _hasStellarUsdc = false;
+  bool get hasStellarUsdc => _hasStellarUsdc;
+
+  @PostConstruct(preResolve: true)
+  Future<void> init() async {
```

Comment:
I don't think it should be "blocking" call preventing from accessing the application. Imagine that user has bad internet access, so the app would be initializing pretty long. There should be no long calls in dependency initialization.

Also, I would rather have this checking fully asynchronous, and when the check is done, and user has some money in Stellar account (and they didn't initiate it in this instance of the app), we can show some notification.

Reply:
@ookami-kb I've updated the code. Shouldn't be blocking now and will show screen asynchronously